src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -210,7 +211,13 @@ public final Connection getConnection(final long hardTimeout) throws SQLExceptio
       }
 
       logPoolState("Timeout failure ");
-      throw new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedTimeMs(startTime), lastConnectionFailure.getAndSet(null)));
+      final Throwable cause = lastConnectionFailure.getAndSet(null);
+      final SQLTimeoutException exception = new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedTimeMs(startTime)));
+
+      if (cause != null) {
```

Comment:
I don't understand the benefit of this.  Either way, if the cause is null the result is same, but the current code is simpler.